jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java

```diff
@@ -575,746 +599,144 @@ public void createGlobalDirOnNewProject( @Observes NewProjectEvent newProjectEve
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "passwordfield.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "radiobutton.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "script.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "submitbutton.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "table.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "textarea.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "textfield.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "themes.fw")),
+                            "");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "unorderedlist.fw")),
+                            "");
+
+            if (servletContext != null) {
+
+                // default custom editors
+                try {
+                    String defaultCustomEditorsJSON = servletContext.getRealPath(designerContext + "/defaults/customeditors.json");
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "customeditors.json")),
+                                    getBytesFromFile(new File(defaultCustomEditorsJSON)));
+                } catch (Exception e) {
+                    logger.error("Unable to create default custom editors.");
+                }
+
+                // default color themes
+                try {
+                    String defaultColorThemesJSON = servletContext.getRealPath(designerContext + "/defaults/themes.json");
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "themes.json")),
+                                    getBytesFromFile(new File(defaultColorThemesJSON)));
+                } catch (Exception e) {
+                    logger.error("Unable to create default color themes.");
+                }
+
+                // default icons
+                try {
+                    String defaultEmailIcon = servletContext.getRealPath(designerContext + "/defaults/defaultemailicon.gif");
+                    String defaultLogIcon = servletContext.getRealPath(designerContext + "/defaults/defaultlogicon.gif");
+                    String defaultServiceIcon = servletContext.getRealPath(designerContext + "/defaults/defaultservicenodeicon.png");
+
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "defaultemailicon.gif")),
```

Comment:
Could you please extract this "copy resource to ioservice" functionality to separate function?

You could make this much more readable by then just calling
```
List<String> iconFileNames = Arrays.asList("defaultemailincon.gif", "defautlogicon.gif", ...)
iconFileNames.forEach(this::copyToGlobalsDir);
```